service/service.go

```diff
@@ -35,14 +37,19 @@ func New(conf *config.Config, oracleAccount *panacea.OracleAccount) (*Service, e
 		return nil, fmt.Errorf("failed to set self-enclave info: %w", err)
 	}
 
+	queryClient, err := panacea.LoadQueryClient(context.Background(), conf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load query client: %w", err)
+	}
+
 	grpcClient, err := panacea.NewGrpcClient(conf)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create a new gRPC client: %w", err)
 	}
 
 	subscriber, err := event.NewSubscriber(conf.Panacea.RPCAddr)
 	if err != nil {
-		// TODO: close grpcClient
+		_ = grpcClient.Close()
```

Comment:
If an error occurs while closing grpcClient, how about outputting a warning error here?

Reply:
Should we `panic(err)` when error is not nil from closing gRPC client?

Reply:
I think it would be better to print an error log rather than panic so that you can see what kind of error occurred.

```suggestion
		if err := grpcClient.Close(); err != nil {
		    log.Warn(err)
		}
```

Reply:
I think we may have to close query client as well.

Reply:
Ok. I see. I'll make changes as you suggested!

Reply:
> I think we may have to close query client as well.

Agree.

Reply:
Wait, do we have to close here?
Actually the caller deferred close the service, so all the resources will close.

Reply:
I think `queryClient` and `grpcClient` is closed at `svc.Close()`. Is it right?

Reply:
Yes that's what I thought. So I think we don't have to close each resource when error occurred.
The caller should close the service when the caller creates a new service.
Am I right @gyuguen?

Reply:
Yes. However, since the place where the service is created is implemented like this, it seems that the order needs to be changed.

```go
svc, err := service.New(conf, oracleAccount)
	if err != nil {
		return fmt.Errorf("failed to create service: %w", err)
	}
	defer svc.Close()
```

Reply:
If some subcomponents aren't initialized successfully in the `service.New()`, subcomponents that were already initialized in the `service.New()` must be closed explicitly as Gyuguen said.

Also, an error from any `Close()` function shouldn't stop any remaining tasks. It is just good enough to print an WARN log. After that, we should keep going for remaining tasks, such as closing other subcomponents or returning an error to the caller.
Errors from `Close()` are usually from closing TCP sockets which were already closed for some reason, from disk I/O error due to disk faults, or from deadlocks in the subcomponents. Those errors are generally unrecoverable. It means that there is nothing for us to do with those errors. In fact, most of errors from `Close()` are not critical in general.

Reply:
I got it, thank you.
When error occurs in creating new service, the resources are explicitly closed as you said.